iron-overlay-manager.html

```diff
@@ -11,201 +11,186 @@
 <link rel="import" href="../polymer/polymer.html">
 
 <script>
-
-  /**
-   * @struct
-   * @constructor
-   */
-  Polymer.IronOverlayManagerClass = function() {
-    this._overlays = [];
+(function() {
+  // Singleton instance.
+  // This is compiler-friendly, because properties can be safely renamed.
+  var Manager = {
+    _overlays: [],
     // Used to keep track of the last focused node before an overlay gets opened.
-    this._lastFocusedNodes = [];
+    _lastFocusedNodes: [],
+
+    // iframes have a default z-index of 100, so this default should be at least that.
+    _minimumZ: 101,
+    _backdrops: [],
+    // Memoized backdrop element.
+    _backdropElement: null,
 
     /**
-     * iframes have a default z-index of 100, so this default should be at least
-     * that.
-     * @private {number}
+     * The shared backdrop element.
+     * @returns {Node} backdropElement
      */
-    this._minimumZ = 101;
-
-    this._backdrops = [];
-
-    this._backdropElement = null;
-    Object.defineProperty(this, 'backdropElement', {
-      get: function() {
-        if (!this._backdropElement) {
-          this._backdropElement = document.createElement('iron-overlay-backdrop');
-        }
-        return this._backdropElement;
-      }.bind(this)
-    });
+    get backdropElement() {
+      if (!this._backdropElement) {
+        this._backdropElement = document.createElement('iron-overlay-backdrop');
+      }
+      return this._backdropElement;
+    },
 
     /**
      * The deepest active element.
-     * returns {?Node} element the active element
+     * @returns {Node|undefined} activeElement the active element
      */
-    this.deepActiveElement = null;
-    Object.defineProperty(this, 'deepActiveElement', {
-      get: function() {
-        var active = document.activeElement;
-        // document.activeElement can be null
-        // https://developer.mozilla.org/en-US/docs/Web/API/Document/activeElement
-        while (active && active.root && Polymer.dom(active.root).activeElement) {
-          active = Polymer.dom(active.root).activeElement;
-        }
-        return active;
-      }.bind(this)
-    });
-  };
-
-  /**
-   * If a node is contained in an overlay.
-   * @private
-   * @param {Node} node
-   * @returns {Boolean}
-   */
-  Polymer.IronOverlayManagerClass.prototype._isChildOfOverlay = function(node) {
-    while (node && node !== document.body) {
-      // Use logical parentNode, or native ShadowRoot host.
-      node = Polymer.dom(node).parentNode || node.host;
-      // Check if it is an overlay.
-      if (node && node.behaviors && node.behaviors.indexOf(Polymer.IronOverlayBehaviorImpl) !== -1) {
-        return true;
+    get deepActiveElement() {
```

Comment:
if getters are in all our supported browsers, then why not just change the defineProperties to getters? Much less of a change and gets rid of the literals.